core/src/test/scala/org/apache/spark/io/ZStandardBenchmark.scala

```diff
@@ -101,4 +102,28 @@ object ZStandardBenchmark extends BenchmarkBase {
       }
     }
   }
+
+  private def parallelCompressionBenchmark(): Unit = {
+    val N = 128
+    val data = (1 until 128 * 1024 * 1024).toArray
```

Comment:
This is `Array[Int]`. I guess you wanted to make 128MB size, but this looks like bigger than 512MB.